### PR TITLE
add '| bool' on many conditionals

### PR DESCRIPTION
**Pull request by DO1JLR**
By running ceph-ansible there are a lot ``[DEPRECATION WARNING]`` like these:
```
[DEPRECATION WARNING]: evaluating containerized_deployment as a bare variable,
this behaviour will go away and you might need to add |bool to the expression
in the future. Also see CONDITIONAL_BARE_VARS configuration toggle.. This
feature will be removed in version 2.12. Deprecation warnings can be disabled
by setting deprecation_warnings=False in ansible.cfg.
```

Now appended ``| bool`` on a lot of the affected variables.

Sometimes the coding style from ``variable|bool`` changed to ``variable | bool`` *(with spaces at the pipe)*.

I hope you like it. If I should change something or did something completly wrong... Please say it!

### CODE REVIEW

**Comment by guits** [comment]
@DO1JLR Thanks for this PR.

Could you add a sign off in your commit please?

**Comment by guits** [comment]
Could you add the `Closes: #4022` keyword in your commit message as well?

Thanks!

**Comment by DO1JLR** [comment]
> Could you add a sign off in your commit please?
 I added ``Signed-off-by: L3D <l3d@c3woc.de>``. I hope that's alright!

> Could you add the Closes: #4022 keyword in your commit message as well?
I added it..

I hope that's all right or did I miss something?

**Comment by guits** [comment]
@DO1JLR looks good to me, let's wait for the CI

Thanks

File: roles/ceph-handler/handlers/main.yml [hunk 1]
```diff
@@ -48,10 +48,10 @@
       listen: "restart ceph mons"
       when:
         # We do not want to run these checks on initial deployment (`socket.rc == 0`)
-        - mon_group_name in group_names
-        - containerized_deployment
+        - mon_group_name in group_names | bool
```
Comment (guits) [inline roles/ceph-handler/handlers/main.yml]:
```suggestion
        - mon_group_name in group_names
```

**Comment by DO1JLR** [comment]
I rebased the pull request with the current master and did the change @guits requested.

File: roles/ceph-handler/handlers/main.yml (hunk 1 above)
Comment (DO1JLR) [reply roles/ceph-handler/handlers/main.yml]: Upsi, you are right.

File: roles/ceph-handler/handlers/main.yml [hunk 2]
```diff
@@ -48,10 +48,10 @@
       listen: "restart ceph mons"
       when:
         # We do not want to run these checks on initial deployment (`socket.rc == 0`)
-        - mon_group_name in group_names
-        - containerized_deployment
+        - mon_group_name in group_names
```
Comment (dsavineau) [inline roles/ceph-handler/handlers/main.yml]: trailling space

File: roles/ceph-handler/handlers/main.yml [hunk 3]
```diff
@@ -48,10 +48,10 @@
...
         # We do not want to run these checks on initial deployment (`socket.rc == 0`)
-        - mon_group_name in group_names
-        - containerized_deployment
+        - mon_group_name in group_names 
+        - containerized_deployment | bool
         - ceph_mon_container_stat.get('rc') == 0
-        - hostvars[item]['_mon_handler_called'] | default(False)
+        - hostvars[item]['_mon_handler_called'] | default(False) | bool
         - ceph_mon_container_stat.get('stdout_lines', [])|length != 0
         - not rolling_update
```
Comment (dsavineau) [inline roles/ceph-handler/handlers/main.yml]: missing filter

File: roles/ceph-infra/tasks/configure_firewall.yml [hunk 4]
```diff
@@ -8,7 +8,7 @@
   check_mode: no
   changed_when: false
   tags: firewall
-  when: not containerized_deployment
+  when: not containerized_deployment | bool
 
 - when: (firewalld_pkg_query.get('rc', 1) == 0
       or is_atomic)
```
Comment (dsavineau) [inline roles/ceph-infra/tasks/configure_firewall.yml]: missing filter

File: tests/functional/rgw_multisite.yml [hunk 5]
```diff
@@ -21,13 +21,13 @@
...
-      when: not is_atomic
+      when: not is_atomic | bool
 
     - name: generate and upload a random 10Mb file - containerized deployment
       command: >
         docker run --rm --name=rgw_multisite_test --entrypoint=/bin/bash {{ ceph_docker_registry }}/{{ ceph_docker_image }}:{{ ceph_docker_image_tag }} -c 'dd if=/dev/urandom of=/tmp/testinfra.img bs=1M count=10; {{ s3cmd_cmd }} mb s3://testinfra; {{ s3cmd_cmd }} put /tmp/testinfra.img s3://testinfra'
       when:
-        - rgw_zonemaster
+        - rgw_zonemaster | bool
         - containerized_deployment | default(False)
```
Comment (dsavineau) [inline tests/functional/rgw_multisite.yml]: missing filter

File: roles/ceph-handler/handlers/main.yml (hunk 2 above)
Comment (DO1JLR) [reply roles/ceph-handler/handlers/main.yml]: Sorry, I removed the space and force-pushed this change.

File: roles/ceph-handler/handlers/main.yml (hunk 3 above)
Comment (DO1JLR) [reply roles/ceph-handler/handlers/main.yml]:
```
- not rolling_update | bool
```
| bool filter is added.

File: roles/ceph-infra/tasks/configure_firewall.yml (hunk 4 above)
Comment (DO1JLR) [reply roles/ceph-infra/tasks/configure_firewall.yml]:
```
- when: (firewalld_pkg_query.get('rc', 1) == 0
      or is_atomic | bool)
```
is_atomic has now a boolean filter too!

File: tests/functional/rgw_multisite.yml (hunk 5 above)
Comment (DO1JLR) [reply tests/functional/rgw_multisite.yml]:
```
        - containerized_deployment | default(False) | bool
```
How could I miss this?
Now the filter is there too and the change is force pushed and amended in this commit here.

**Comment by DO1JLR** [comment]
> some missing bool filter.
> You can take a look to #3830 to compare both PR.

Thanks @dsavineau 
I added the missing filter in my commit.

But I do not really understand why this pull request might now be merged, but # 3830 was closed.

**Comment by guits** [comment]
@DO1JLR #3830 was opened a bit too early and had to be rebased on master too often because of the number of changes it had obviously.

**Comment by guits** [comment]
jenkins test pipeline

**Comment by DO1JLR** [comment]
I can see jenkins failed with the [ceph-ansible-prs-dev-centos-container-dashboard](https://2.jenkins.ceph.com/job/ceph-ansible-pipeline/81/)  Test.
But I can't identify what exactly the error is. Sorry.

@guits, Maybe you can help or tell me what went wrong?

**Comment by guits** [comment]
@DO1JLR sorry for the late reply, the failure in ceph-ansible-prs-dev-centos-container-dashboard was unrelated to your patch.

could you rebase + fix conflicting files? it should be good to go after that.
Thanks!

**Comment by DO1JLR** [comment]
> @DO1JLR sorry for the late reply, the failure in ceph-ansible-prs-dev-centos-container-dashboard was unrelated to your patch.
> 
> could you rebase + fix conflicting files? it should be good to go after that.
> Thanks!

Thank YOU for your great work wirh ceph-ansible and your patience with me.

I fixed the conflicting files and rebased it.

**Comment by DO1JLR** [comment]
I made a mistake in the ``roles/ceph-infra/tasks/configure_firewall.yml`` file and force-pushed the correction.

**Comment by DO1JLR** [comment]
Thanks :heart:

**Pull request merged**